torch/_dynamo/eval_frame.py

```diff
@@ -293,6 +295,32 @@ def make_set_enable_dynamic(enable: bool):
         )
 
 
+# A thread local storage that serves to store information as Dynamo traces
+# through a user provided function.
+class DynamoTLS(threading.local):
+    # Each string is a summary of a frame Dynamo attempted to trace, stored in
+    # temporal order.
+    traced_frame_infos: List[str] = []
+
+
+dynamo_tls = DynamoTLS()
+
+
+@atexit.register
+def _log_traced_frames():
+    """
+    At program exit, log all of the frames Dynamo has attempted to trace from,
+    excluding the continuation frames generated by Dynamo.
+    """
+    msg = "TorchDynamo attempted to trace the following frames: ["
+    if len(dynamo_tls.traced_frame_infos) > 0:
+        msg += "\n"
+        for traced_frame_info in dynamo_tls.traced_frame_infos:
+            msg += "  * " + traced_frame_info + "\n"
+    msg += "]"
```

Comment:
Use a comprehension?

```python
msg = f"TorchDynamo... [\n{textwrap.indent('\n'.join(dynamo_tls.traced_frame_infos), '  * ')}\n]"
```

Reply:
That's cool but I feel like it hurts readability and modifiability quite a bit.

Reply:
You could break up the comprehension into a few lines, e.g.
```python
msg = '\n'.join(dynamo_tls.trace_frame_infos)
msg = textwrap.indent(msg, '  * ')
msg = f"TorchDynamo... [\n{msg}\n]"
```

Reply:
I'll take that, updated.